sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubProducerClient.cs

```diff
@@ -59,88 +67,257 @@ public class EventHubProducer : IAsyncDisposable
-        ///   owned by this instance and are safe from changes made by consumers.  It is considered the responsibility of the
-        ///   caller to ensure that any needed cloning of options is performed.
+        ///   If the connection string is copied from the Event Hub itself, it will contain the name of the desired Event Hub,
+        ///   and can be used directly without passing the <paramref name="eventHubName" />.  The name of the Event Hub should be
+        ///   passed only once, either as part of the connection string or separately.
         /// </remarks>
         ///
-        internal EventHubProducer(TransportEventHubProducer transportProducer,
-                                  Uri endpoint,
-                                  string eventHubName,
-                                  EventHubProducerOptions producerOptions,
-                                  EventHubRetryPolicy retryPolicy)
+        public EventHubProducerClient(string connectionString,
+                                      string eventHubName,
+                                      EventHubProducerClientOptions producerOptions)
         {
-            Argument.AssertNotNull(transportProducer, nameof(transportProducer));
-            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Argument.AssertNotNullOrEmpty(connectionString, nameof(connectionString));
+            producerOptions = producerOptions?.Clone() ?? new EventHubProducerClientOptions();
+
+            OwnsConnection = true;
+            Connection = new EventHubConnection(connectionString, eventHubName, producerOptions.ConnectionOptions);
+            PartitionId = producerOptions.PartitionId;
+            RetryPolicy = producerOptions.RetryOptions.ToRetryPolicy();
+            InnerProducer = Connection.CreateTransportProducer(producerOptions);
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventHubProducerClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace to connect to.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub to associated the connection with.</param>
+        /// <param name="credential">The Azure managed identity credential to use for authorization.  Access controls may be specified by the Event Hubs namespace or the requested Event Hub, depending on Azure configuration.</param>
+        /// <param name="producerOptions">A set of options to apply when configuring the producer.</param>
+        ///
+        public EventHubProducerClient(string fullyQualifiedNamespace,
+                                      string eventHubName,
+                                      TokenCredential credential,
+                                      EventHubProducerClientOptions producerOptions = default)
```

Comment:
I don't get why we don't do the same for other constructors (letting `options = default` instead of having two constructors).

Reply:
The goal was to avoid having default parameters in the extreme "hello world" constructors (via the architecture board), but there is also a concern of having too many overloads.  The recommendation was to make the connection string versions have the overload and keep more advanced constructors with `default` to limit the number of overloads.

There is already concern over how many overloads because of the `eventHub` + `connectionString` variants.